### PR TITLE
Disable virtual threads by default on generated Spring app

### LINKED ISSUE #10474: Disable virtual threads by default on generated Spring app
There are some edge cases where the usage of java 21 virtual threads is causing issues, so it's better to turn it off, as done by Spring Boot.

### PR DESCRIPTION
**Pull request by Harsh4902**
- turned off virtual thread in by changing `spring.threads.virtual.enabled` to `false` in `application.yaml`
- Fixes #10474

### CODE REVIEW

File: src/main/resources/config/application.yml [hunk 1]
```diff
@@ -78,7 +78,7 @@ server:
 spring:
   threads:
     virtual:
-      enabled: true
+      enabled: false
```
Comment (DamnClin) [inline src/main/resources/config/application.yml:81]: This is the configuration for the generator (the app you start locally). The goal here was to disable that for generated applications so update in `SpringBootCoreModuleFactory`

Comment (Harsh4902) [reply src/main/resources/config/application.yml]: Changed value of  `spring.threads.virtual.enabled` to `false` in `SpringBootCoreModuleFactory`.

File: src/main/java/tech/jhipster/lite/generator/server/springboot/core/domain/SpringBootCoreModuleFactory.java [hunk 2]
```diff
@@ -78,7 +78,7 @@ public JHipsterModule buildModule(JHipsterModuleProperties properties) {
         .and()
       .springMainProperties()
         .set(propertyKey("spring.application.name"), propertyValue(baseName))
-        .set(propertyKey("spring.threads.virtual.enabled"), propertyValue(true))
+        .set(propertyKey("spring.threads.virtual.enabled"), propertyValue(false))
```
Comment (murdos) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/core/domain/SpringBootCoreModuleFactory.java:81]: In fact we should just remove the property, in order to use default Spring boot behaviour

**Review by murdos — CHANGES_REQUESTED** [review]

File: src/main/java/tech/jhipster/lite/generator/server/springboot/core/domain/SpringBootCoreModuleFactory.java (hunk 2 above)
Comment (Harsh4902) [reply src/main/java/tech/jhipster/lite/generator/server/springboot/core/domain/SpringBootCoreModuleFactory.java:81]: Oh, ok.

**Comment by Harsh4902** [comment]
@murdos I am closing this PR, will raise a fresh PR with minimal commit.

**Pull request closed without merging**